File: app/src/main/java/ir/ghararemaghzha/game/classes/MySettingsPreference.java
package ir.ghararemaghzha.game.classes;

import android.content.Context;
import android.content.SharedPreferences;
import androidx.security.crypto.EncryptedSharedPreferences;
import androidx.security.crypto.MasterKey;
import java.io.IOException;
import java.security.GeneralSecurityException;

public class MySettingsPreference {

    private static MySettingsPreference instance;
    private static SharedPreferences sharedPreferences;




    public static MySettingsPreference getInstance(Context context){
        if (instance==null) {
            instance = new MySettingsPreference(context);
        }

        return instance;
    }

    private MySettingsPreference(Context context){
        try {
            MasterKey masterKey = new MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build();
            sharedPreferences = EncryptedSharedPreferences.create(
                    context,
                    "settings_preference",
                    masterKey,
                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM);
        } catch (GeneralSecurityException | IOException e) {
            e.printStackTrace();
        }
    }
    public void setMusic(boolean state){
        sharedPreferences.edit().putBoolean("music",state).apply();
    }
    public boolean getMusic(){
        return  sharedPreferences.getBoolean("music",true);
    }

    public void setNotification(boolean state){
        sharedPreferences.edit().putBoolean("Notification",state).apply();
    }
    public boolean getNotification(){
        return  sharedPreferences.getBoolean("Notification",true);
    }

    public void setAutoNext(boolean state){
        sharedPreferences.edit().putBoolean("AutoNext",state).apply();
    }
    public boolean getAutoNext(){
        return  sharedPreferences.getBoolean("AutoNext",false);
    }

}

File: app/src/main/java/ir/ghararemaghzha/game/classes/MySettingsPreference.kt
package ir.ghararemaghzha.game.classes

import android.content.Context
import androidx.core.content.edit
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey

class MySettingsPreference private constructor(ctx: Context) {
    private val masterKey = MasterKey.Builder(ctx).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
    private val sp = EncryptedSharedPreferences.create(
            ctx,
            "settings_preference",
            masterKey,
            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM)


    companion object {
        @Volatile
        private var instance: MySettingsPreference? = null

        fun getInstance(ctx: Context): MySettingsPreference =
                instance ?: synchronized(this) {
                    instance ?: MySettingsPreference(ctx).also { instance = it }
                }
    }

    fun setMusic(state: Boolean) = sp.edit { putBoolean("music",state) }

    fun getMusic() = sp.getBoolean("music", true)

    fun setNotification(state: Boolean) = sp.edit {putBoolean("Notification", state) }

    fun getNotification() = sp.getBoolean("Notification", true)

    fun setAutoNext(state: Boolean) = sp.edit{putBoolean("AutoNext", state)}

    fun getAutoNext() = sp.getBoolean("AutoNext", false)


}

